onie-mk-demo.sh

```diff
@@ -93,6 +93,7 @@ sed -i -e "s/%%DEMO_TYPE%%/$demo_type/g" \
        -e "s/%%ONIE_IMAGE_PART_SIZE%%/$onie_image_part_size/" \
        -e "s/%%EXTRA_CMDLINE_LINUX%%/$EXTRA_CMDLINE_LINUX/" \
        -e "s@%%OUTPUT_RAW_IMAGE%%@$output_raw_image@" \
+       -e "s/%%IMAGE_ASIC%%/$machine/g" \
     $tmp_installdir/install.sh || clean_up 1
```

Comment:
>install.sh [](http://example.com/codeflow?start=20&length=10)

You don't need to embed the info into this file. $machine is already in the file `machine.conf`, which is in the same directory of `install.sh` during installation.

Reply:
I saw the statement is 
` if [ -r machine.conf] . ./machine.conf `
so we cannot guarantee the machine.conf is always available,right? 
otherwise we can skip the if statement.

Reply:
The Line 104~105 will guarantee the file exists in onie image. The code you mention should be fixed.